hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/AbstractGenericRecordBuilder.java

```diff
@@ -129,6 +133,14 @@ public GenericRecordBuilder setTimestampWithTimezone(@Nonnull String fieldName,
     @Nonnull
     @Override
     public GenericRecordBuilder setArrayOfGenericRecord(@Nonnull String fieldName, @Nullable GenericRecord[] value) {
+        if (value != null) {
+            for (GenericRecord genericRecord : value) {
```

Comment:
Hmm, so we have to traverse the array here to perform the checks. Not ideal, but I couldn't see what else we can do to detect it early. So, I believe it is OK 👍

Reply:
I don't think there's any other way